lib/ActivityStreams.js

```diff
@@ -96,25 +96,26 @@ ActivityStreams.prototype = {
    */
   _respondToPlacesRequests(msgName, params) {
     let {msg, worker} = params;
+    const append = msg.meta && msg.meta.append;
```

Comment:
The memoization key generator will not handle this correctly.

I would suggest to flatten the `msg` object by adding `append` as a parameter of `msg` and removing `meta` altogether, or to improve the cache key generation.